lib/common/engine.js

```diff
@@ -10,10 +11,18 @@ const defaultOptions = {
   stateCacheSize: STATE_CACHE_SIZE,
 };
 
+function factory(name) {
```

Comment:
A whole file for a single function whose only purpose is to import a few other files?